torch/_inductor/autotuner/model.py

```diff
@@ -0,0 +1,801 @@
+)
+from torch._inductor.virtualized import V
+from torch._inductor.coordinate_descent_tuner import CoordescTuner, get_field, set_field
+from torch._inductor.utils import sympy_product, get_dtype_size
+
+log = logging.getLogger(__name__)
+
+
+### Raw data related
+class AutotunerRawData(NamedTuple):
+    io_deps: Tuple[
+        List[dependencies.Dep], List[dependencies.Dep], List[int], List[int], List[int]
+    ]
+    read_write: dependencies.ReadWrites
+    src_code: str
+    autotuner_dict: Dict
+
+    def __repr__(self):
+        return f"AutotunerRawData({self.io_deps}, {self.read_write}, {self.src_code}, {self.autotuner_dict})"
+
+
+def get_reads_writes(cur_scheduler, node, src_code):
+    if isinstance(node, scheduler.NopKernelSchedulerNode):
+        return 0
+    reads = {dep.name for dep in node.read_writes.reads}
+    writes = {dep.name for dep in node.read_writes.writes}
+
+    def is_materialized(buf):
+        buf_uses = {user.node for user in cur_scheduler.name_to_node[buf].users}
+        return len(buf_uses - set(node.snodes)) > 0
+
+    if isinstance(node, scheduler.FusedSchedulerNode):
+        removed_buffers = {dep for dep in writes if not is_materialized(dep)}
+        writes = writes - removed_buffers
+        reads = reads - removed_buffers
+
+    dep_removed = list()
+    strides = list()
+    sizes = list()
+    node_bytes = list()
+
+    # To have deterministic order
+    def f(dep_set):
+        for dep in sorted(list(dep_set), key=lambda x: x.name):
+            if dep.name not in reads | writes:
+                continue
+            if dep.name in V.graph.name_to_buffer:
+                buf = V.graph.name_to_buffer[dep.name]
+            elif dep.name in V.graph.graph_inputs:
+                buf = V.graph.graph_inputs[dep.name]
+            else:
+                continue
+
+            dep_removed.append(dep)
+            if isinstance(dep, (StarDep, WeakDep)):
+                strides.append(None)
+                sizes.append(None)
+                node_bytes.append(
+                    V.graph.sizevars.size_hint(sympy_product(buf.get_size()))
+                    * get_dtype_size(buf.get_dtype())
+                )
+            else:
+                strides.append(V.graph.sizevars.stride_hints(dep.index, dep.var_names))
+                sizes.append([V.graph.sizevars.size_hint(s) for s in dep.size])
+                node_bytes.append(
+                    V.graph.sizevars.size_hint(sympy_product(dep.size))
+                    * get_dtype_size(buf.get_dtype())
+                )
+
+    f(node.read_writes.reads)
+    read_len = len(dep_removed)
+    f(node.read_writes.writes)
+    assert len(dep_removed) == len(node_bytes) == len(strides) == len(sizes)
+    return (
+        dep_removed[:read_len],
+        dep_removed[read_len:],
+        strides,
+        sizes,
+        node_bytes,
+    )
+
+
+### feature extraction related
+
+# op_dict needs to be deterministic
+OP_DICT = {
```

Comment:
Need a way to keep this updated and/or check that is up to date

Reply:
Currently it will throw if it encounters unknown op:
https://github.com/pytorch/pytorch/pull/107488/files#diff-351bdd9a44612eb5097abb55e76cdfd58f1871cf1004ce6aae82a0077b2194f5R751

Likely a way to notify people to update.